docs/adr/0024-upgrade-templating-toolset-of-auth-server-emails.md

```diff
@@ -0,0 +1,261 @@
+# Upgrade the templating toolset of Auth Server emails
+
+- Status: proposed
+- Deciders: Yogita Bhatia, Lauren Zugai, Jody Heavener, Vijay Budhram, [TBD]
+- Date: 2021-06-04
+
+Technical Story: [#8600](https://github.com/mozilla/fxa/issues/8600)
+
+## Context and Problem Statement
+
+[Mustache](https://www.npmjs.com/package/mustache), used in `fxa-auth-server` email templates, can accommodate very limited logic in its templates which has been a pain point for Firefox Accounts engineers and discussion around using a different templating system [began here](https://github.com/mozilla/fxa/issues/4627). While converting our emails to a more modern templating solution, there is an opportunity to evaluate what stack would be the most ideal for FxA emails beyond a proposed templating solution. This includes evaluating our CSS options, improving how we can preview emails in various states, and our localization tools, and the best approach for landing new templates in production.
```

Comment:
To be exact, we're using [Handlebars](https://www.npmjs.com/package/handlebars) - which is an extension of Mustache and [adds some important features](https://github.com/handlebars-lang/handlebars.js#differences-between-handlebarsjs-and-mustache) that [we use for email templates](https://github.com/mozilla/fxa/blob/main/packages/fxa-auth-server/lib/senders/templates/index.js#L33-L38).

Reply:
@Yogita98 I think we can address Les' comment here by updating the first sentence:

```
[Handlebars](https://www.npmjs.com/package/mustache), an extension of [Mustache](https://www.npmjs.com/package/mustache) ([see the differences](https://github.com/handlebars-lang/handlebars.js#differences-between-handlebarsjs-and-mustache)) used in `fxa-auth-server` email templates, can accommodate very limited logic in its templates which has been a pain point for Firefox Accounts engineers and discussion around using a different templating system [began here](https://github.com/mozilla/fxa/issues/4627).
```